### ISSUE TITLE
`Select` component fails eslint a11y rule

### ISSUE THREAD

**Issue by johnpmitsch** [patternfly/patternfly-react#5042]
**Describe the issue. What is the expected and unexpected behavior?**
The `Select` component fails a11y eslint rule for nested inputs:

`error  Form label must have ALL of the following types of associated control: nesting, id  jsx-a11y/label-has-for`

The rule can be found [here](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/master/docs/rules/label-has-for.md)

**Please provide the steps to reproduce. Feel free to link CodeSandbox or another tool.**

versions:
```
eslint@6.8.0
@patternfly/patternfly@4.50.4
@patternfly/react-core@4.63.3
```

Here is an example of a nested `Select` inside a `label`
```jsx
        <label htmlFor={`select ${title}`}>
          {`${title}:`}
            <Select
              id={`select ${title}`}
              key="type-dropdown"
              variant={SelectVariant.single}
              onToggle={onToggle}
              onSelect={onSelect}
              selections={selected}
              isOpen={isOpen}
            >
              {selectItems}
            </Select>
        </label>
```
**Is this a bug or enhancement? If this issue is a bug, is this issue blocking you or is there a work-around?**
This is a bug. I tried `FormSelect` as well and it still failed even though it was nested. I'm not sure if the rule wants it to be a direct child of the label in the rendered HTML? The workaround is to disable the `nesting` part of the rule, just enforce `id`.

**What is your product and what release version are you targeting?**
any

**Comment by tlabaj** [comment]
cc @jessiehuff

**Comment by nicolethoen** [comment]
Hi @johnpmitsch 
I just dug up this issue to take another look.
It looks like the a11y rule you linked as been replaced with this rule: [label-has-associated-control](https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/master/docs/rules/label-has-associated-control.md).

In summary, to make a valid select with a label, you can do one of a few things, two of which include:

1. Wrap the input in a label (with no `htmlFor` prop)
2. Use the `htmlFor` prop when input and label are siblings

I'll also note that when you have a space (' ') in your `id` which may be impacting the relationship between the `htmlFor` attribute and the input's id. [See docs here](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/id).

I believe you should be able to solve your linting issues if you be sure to follow all these guidelines. Let me know if this is not the case.
